sources/packages/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -448,4 +456,120 @@ export class ApplicationStudentsController extends BaseController {
       applicationNumber: application.applicationNumber,
     };
   }
+
+  /**
+   * Get in progress details of an application by application id.
+   * @param applicationId application id.
+   * @returns application details.
+   */
+  @Get(":id/in-progress")
+  @ApiNotFoundResponse({
+    description: "Application id not found.",
+  })
+  async getInProgressApplicationDetails(
+    @Param("id", ParseIntPipe) applicationId: number,
+  ): Promise<InProgressApplicationDetailsAPIOutDTO> {
+    const application = await this.applicationService.getApplicationByIdAndUser(
```

Comment:
We must check if the student has access to the application and we should not rely on the userId since the same student can have multiple ones.

Reply:
Where are we using the userId here @andrewsignori-aot ?

Reply:
@dheepak-aot we are not using `userId` here and this is the first problem, there is no check if the student has access to the application.
@ann-aot this PR comment can be fixed by adding the userId which will ensure the authorization check in the same way as the rest of the controller.
Ideally, we will have a tech-debit to change the authorization check to be based on the student.